code/game/objects/items/weapons/RCD.dm

```diff
@@ -382,3 +388,278 @@ RCD
+	flags = CONDUCT | NOBLUDGEON
+	force = 0
+	throwforce = 10
+	throw_speed = 3
+	throw_range = 5
+	w_class = WEIGHT_CLASS_NORMAL
+	materials = list(MAT_METAL=100000)
+	origin_tech = "engineering=4;materials=2"
+	req_access_txt = "11"
+	armor = list(melee = 0, bullet = 0, laser = 0, energy = 0, bomb = 0, bio = 0, rad = 0, fire = 100, acid = 50)
+	resistance_flags = FIRE_PROOF
+	var/datum/effect_system/spark_spread/spark_system
+	var/matter = 200
+	var/max_matter = 200
+	var/working = 0
+	var/mode = 1
+	actions_types = list(/datum/action/item_action/pick_color)
+
+	var/wallcost = 10
+	var/floorcost = 15
+	var/launchcost = 5
+	var/deconcost = 10
+
+	var/walldelay = 10
+	var/floordelay = 10
+	var/decondelay = 15
+
+	var/sheetmultiplier	= 4
+
+	var/color_choice = null
+
+	var/no_ammo_message = "<span class='warning'>The \'Low Ammo\' light on \
+		the RLD blinks yellow.</span>"
+
+/obj/item/weapon/rld/ui_action_click(mob/user, var/datum/action/A)
+	if(istype(A, /datum/action/item_action/pick_color))
+		color_choice = input(usr,"Choose Color") as color
+	else
+		..()
+
+
+/obj/item/weapon/rld/New()
+	..()
+
+	desc = "An RLD. It currently holds [matter]/[max_matter] matter-units."
+	spark_system = new /datum/effect_system/spark_spread
+	spark_system.set_up(5, 0, src)
+	spark_system.attach(src)
+
+
+/obj/item/weapon/rld/Destroy()
+	qdel(spark_system)
+	spark_system = null
+	. = ..()
+
+/obj/item/weapon/rld/attackby(obj/item/weapon/W, mob/user, params)
+	var/loaded = 0
+	if(istype(W, /obj/item/weapon/rcd_ammo))
+		var/obj/item/weapon/rcd_ammo/R = W
+		if((matter + R.ammoamt) > max_matter)
+			to_chat(user, "<span class='warning'>The RLD can't hold any more matter-units!</span>")
+			return
+		qdel(W)
+		matter += R.ammoamt
+		icon_state = "rld-[round(matter/35)]"
+		update_icon()
+		playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		loaded = 1
+	else if(istype(W, /obj/item/stack/sheet/glass))
+		loaded = loadwithsheets(W, sheetmultiplier, user)
+	if(loaded)
+		to_chat(user, "<span class='notice'>The RLD now holds [matter]/[max_matter] matter-units.</span>")
+		desc = "An RLD. It currently holds [matter]/[max_matter] matter-units."
+	else
+		return ..()
+
+/obj/item/weapon/rld/proc/loadwithsheets(obj/item/stack/sheet/S, value, mob/user)
+	var/maxsheets = round((max_matter-matter)/value)    //calculate the max number of sheets that will fit in RLD
+	if(maxsheets > 0)
+		var/amount_to_use = min(S.amount, maxsheets)
+		S.use(amount_to_use)
+		matter += value*amount_to_use
+		playsound(src.loc, 'sound/machines/click.ogg', 50, 1)
+		to_chat(user, "<span class='notice'>You insert [amount_to_use] [S.name] sheets into the RLD. </span>")
+		icon_state = "rld-[round(matter/35)]"
+		update_icon()
+		return TRUE
+	to_chat(user, "<span class='warning'>You can't insert any more [S.name] sheets into the RLD!")
+	return FALSE
+
+/obj/item/weapon/rld/attack_self(mob/user)
+	//Change the mode
+	playsound(src.loc, 'sound/effects/pop.ogg', 50, 0)
+	switch(mode)
```

Comment:
This might have a more efficent method by having it done in another way but I dont know how.